07-Template-Distribution-Protocol.md

```diff
@@ -168,19 +168,21 @@ the server and nothing needs to be done by client.
 | mining_job_token    | B0_255               | Token that makes the client eligible for committing a mining job for approval/transaction         |
 |                     |                      | negotiation or for identifying custom mining job on mining connection.                            |
 
-## 7.13 `IdentifyTransactions` (Server->Client)
+## 7.13 `IdentifyTransactions` (Client -> Server, Server -> Client)
 
-Sent by the Server in response to a `AllocateTxs.Success` message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.
+Sent by the TP to the JDS in response to a `AllocateTxs` message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.
+Or sent by the JDC to the TP when the JDS need to identify a transaction.
 
 | Field Name          | Data Type            | Description                                                                                       |
 | ------------------- | -------------------- | ------------------------------------------------------------------------------------------------- |
 | mining_job_token    | B0_255               | Token that makes the client eligible for committing a mining job for approval/transaction         |
 |                     |                      | negotiation or for identifying custom mining job on mining connection.                            |
 
 
-## 7.14 `IdentifyTransactions.Success` (Client->Server)
+## 7.14 `IdentifyTransactions.Success` (Client->Server, Server -> Client)
 
-Sent by the Client in response to an `IdentifyTransactions` message to provide the full set of transaction data hashes.
+Sent by the JDS to the TP in response to an `IdentifyTransactions` message to provide the full set of transaction data hashes.
```

Comment:
Could you help me clarify the new logic?:

1. JDS sends `AllocateTxs` to the TP
2. The TP checks its mempool for the received shorttxids
3. Only in the event of collisions, the TP sends `IdentifyTransactions`?
4. JDS responds with `IdentifyTransactions.Success`, it contains all the full tx hashes and the TP can lookup the full tx hash in the mempool for the shorttixds that had the collision?

Alternatively in step 3., if the TP finds missing shorttxids from the mempool, then the TP will send `ProvideMissingTransactions` to receive the full serialized tx data according to the index position of the shorttxids from `AllocateTxs`

5. Finally, if any transactions reconstructed or received from the client is in someway invalid, then the server responds with `AllocateTxs.Error`
else if all of the above passes, server sends `AllocateTxs.Success`?

Reply:
Yes, we have `IdentifyTransactions` for collisions and `ProvideMissingTransactions` for missing txs in the mempool.
Just for a double check @4ss0 do you agree?

Reply:
yes, it make sense to me, Pool can only ask to include one specific transaction that have more fees. Also in the specs there was no mention of other causes than collisions for the ` IdentifyTransactions` message.

Reply:
Pool should not ask to include txs IMO. Just accept what have been passed, assuming that it leads to valid blocks.